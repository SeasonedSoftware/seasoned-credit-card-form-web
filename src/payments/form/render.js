import React from 'react'
import Form from './Form'

export default props => options => <Form {...props} {...options} />
