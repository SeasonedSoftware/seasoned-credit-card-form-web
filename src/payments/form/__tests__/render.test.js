import React from 'react'
import renderer from 'react-test-renderer'

import render from '../render'

jest.mock('../Field', () => props => <div {...props}>Field</div>)

jest.mock('formik', () => ({
  Form: props => <div {...props}>Form</div>,
  Field: props => <div {...props}>Field</div>,
}))

const values = {
  name: '',
  number: '',
  expiry: '',
  cvc: '',
}

const props = {
  id: 1,
  token: 'abababa',
  channel: 'whatsapp',
  text: 'paga nois',
  list: [],
  offer: {
    consumer: {
      fullName: 'foo',
    },
    discount: 50,
    valueToPay: 200,
    originalValue: 250,
    valueWithInterest: 350,
    contractNumber: '123',
  },
}

it('renders correctly', () => {
  const Form = render(props)
  const tree = renderer.create(<Form values={values} />).toJSON()
  expect(tree).toMatchSnapshot()
})
