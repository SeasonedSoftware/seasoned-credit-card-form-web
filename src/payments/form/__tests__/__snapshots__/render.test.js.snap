// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  errors={Object {}}
  handleChange={[MockFunction]}
  language="en-US"
  touched={Object {}}
  values={
    Object {
      "cvc": "",
      "expiry": "",
      "name": "",
      "number": "",
    }
  }
>
  Form
</div>
`;
